### PR TITLE
Attempt to fix cygwin build failure

### PR DESCRIPTION
**Pull request by kevinbackhouse**
I found another project that have been having a similar issue with their cygwin build. They fixed it by installing the packages in a separate step: https://github.com/vsg-dev/VulkanSceneGraph/pull/585. Trying the same approach here.

### CODE REVIEW

**Comment by kevinbackhouse** [comment]
@piponazo: this seems to be only half the solution. The cygwin build is failing due to a problem with `std::filesystem` now.

**Review by neheb — APPROVED** [review]

File: .github/workflows/on_PR_windows_matrix.yml [hunk 1]
```diff
@@ -193,6 +201,7 @@ jobs:
             -DEXIV2_BUILD_UNIT_TESTS=OFF \
             -S . -B build && \
           cmake --build build --parallel
+        shell: C:\tools\cygwin\bin\bash.exe -eo pipefail '{0}'
```
Comment (kmilos) [inline .github/workflows/on_PR_windows_matrix.yml:204]: Not strictly needed, this is already set up as the default shell above...

**Comment by kmilos** [comment]
What about switching to https://github.com/cygwin/cygwin-install-action as some other projects are doing?

**Comment by piponazo** [comment]
> @piponazo: this seems to be only half the solution. The cygwin build is failing due to a problem with `std::filesystem` now.

Cool! I remember we added the file `cmake/FindFilesystem.cmake` to deal in a clean way with old compilers in which we needed to add the `-lstdc++fs` linking flag. For some reason the cygwin build is not passing those cmake tests ... although I would say that cmake file was there since a long time ago (before the cygwin CI job broke).

**Comment by piponazo** [comment]
> What about switching to https://github.com/cygwin/cygwin-install-action as some other projects are doing?

This is what I was trying in another branch. I will spend some time later today on it.

**Comment by kevinbackhouse** [comment]
> What about switching to https://github.com/cygwin/cygwin-install-action as some other projects are doing?

I don't know about that one! Sounds like a good plan.

**Pull request closed without merging**